File: 1128-number-of-equivalent-domino-pairs/1128-number-of-equivalent-domino-pairs.py
class Solution:
    def numEquivDominoPairs(self, dominoes: List[List[int]]) -> int:
        ans, seen = 0, dict()

        for a, b in dominoes:
            seen[(min(a, b), max(a, b))] = seen.get((min(a, b), max(a, b)), 0) + 1

        for (a, b), freq in seen.items():
            ans += freq * (freq - 1) >> 1

        return ans

File: 1128-number-of-equivalent-domino-pairs/1128-number-of-equivalent-domino-pairs.cpp
class Solution {
public:
    int numEquivDominoPairs(vector<vector<int>>& dominoes) {
        int ans = 0;
        unordered_map<string, int> seen;

        for (const auto& domino : dominoes){
            int a = min(domino[0], domino[1]), b = max(domino[0], domino[1]);
            ++seen[to_string(a) + " " + to_string(b)];
        }

        for (const auto& [p, freq] : seen) ans += freq * (freq - 1) >> 1;
        
        return ans;
    }
};